pyccl/halos/halo_model.py

```diff
@@ -82,8 +82,8 @@ def __init__(self, *, mass_function, halo_bias, mass_def=None,
             'log10M_min': log10M_min, 'log10M_max': log10M_max, 'nM': nM,
             'integration_method_M': integration_method_M, 'k_min': k_min}
         self._lmass = np.linspace(log10M_min, log10M_max, nM)
-        self._mass = 10.**self._lmass
-        self._m0 = self._mass[0]
+        self.M = 10.**self._lmass
```

Comment:
Not like this, it has to be a property, otherwise users will be able to modify it, because arrays are mutable, and this could resort to very weird things happening.
I'd suggest returning a copy of the masses. I checked, and copying takes O(100ns), so it's fine.
```python
@property
def M(self):
    return self._M.copy()
```

It was ok when it was private, but now that we expose it we need to be extra careful.

Reply:
Yeah, I hate this. I'll come up with a different way.

Reply:
me too actually, just make `_mass` private again.

Reply:
I went back to everything private and created a single public function of the hmc that allows us to integrate any function of mass.

Reply:
Is this new function any useful, or is it a special-case solution to our problem?

Reply:
It could be generally useful, I think. It's an easy way for users to calculate these integrals, which are quite ubiquitous

Reply:
Yea, I see, so you are allocating the integration work back to `hmc` where it belongs. That's fine.
Can we generalize it in that case? Either create one for mbf or add an argument for users to flag if they want an integral over mf or mbf.

Reply:
fair